crates/wasmtime/src/component/linker.rs

```diff
@@ -161,8 +161,9 @@ impl<T> Linker<T> {
                 .strings
                 .lookup(name)
                 .and_then(|name| self.map.get(&name));
-            cx.definition(ty, import)
-                .with_context(|| format!("import `{name}` has the wrong type"))?;
+            cx.definition(ty, import).with_context(|| {
+                format!("component imports `{name}`, but a correct implementation of `{name}` could not be found")
```

Comment:
```suggestion
                format!("component imports {desc} `{name}`, but a matching implementation was not found", desc = ty.desc())
```

With an additional implementation of `TypeDef::desc` to say what it is, e.g. "instance", "type", "function", etc.